File: web/src/model/victron_controller/dashboard/from_0_2_0_soc-projection.ts
// @ts-nocheck
import {SocProjection as v0_2_0_SocProjection} from './v0_2_0/SocProjection'
import {SocProjection as dashboard_SocProjection} from './SocProjection'

export function convert__soc_projection__from__0_2_0(from: v0_2_0_SocProjection): dashboard_SocProjection {
    return new dashboard_SocProjection (
        JSON.parse(JSON.stringify(from.segments)),
        from.net_power_w,
        from.capacity_wh,
        from.charge_rate_w
    )
}

File: crates/dashboard-model/src/victron_controller/dashboard/from_0_2_0_soc_projection.rs


pub fn convert__soc_projection__from__0_2_0(from: &crate::victron_controller::dashboard::v0_2_0::soc_projection::SocProjection) -> crate::victron_controller::dashboard::soc_projection::SocProjection {
    crate::victron_controller::dashboard::soc_projection::SocProjection {
        segments: serde_json::from_value(serde_json::to_value(&from.segments).unwrap()).unwrap(),
        net_power_w: from.net_power_w.clone(),
        capacity_wh: from.capacity_wh.clone(),
        charge_rate_w: from.charge_rate_w.clone(),
    }
}